brave/brave-core
# feat(wallet): Add Ramp as an Offramp Provider

## Description 
Add's `Ramp` as an `Offramp Provider` in the Wallet

## Supported tokens to sell are:

###  Ethereum Mainnet
- `Ethereum` 
- `Basic Attention Token`
- `The Sandbox`
- `ENS`
- `DAI`
- `Tether`
- `USD Coin`
- `Chainlink`
- `Decentraland`
- `Rally`

###  Binance Smart Chain Mainnet
- `BNB`
- `Binance USD`

###  Celo Mainnet
- `Celo`
- `Celo Dollar`

###  Polygon Mainnet
- `Polygon`
- `Ethereum`
- `USD Coin`
- `DAI Stablecoin`

###  Avalanche Mainnet
- `Avalanche`
- `USD Coin`

###  Arbitrum Mainnet
- `Ethereum`

###  Optimism Mainnet
- `Ethereum`
- `DAI Stablecoin`

###  Fantom Mainnet
- `Fantom`

###  Solana Mainnet
- `Solana`
- `Tether`
- `USD Coin`


Resolves <https://github.com/brave/brave-browser/issues/27872>

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### Testing from Portfolio
1. Go to the `Portfolio` page and click `Basic Attention Token` (on the Ethereum Network).
2. If you have a balance you will see a `Sell` button to the right of your `Account`
3. Click on `Sell` and the `Sell` modal will display.
4. Check that the `HALF` and `MAX` buttons work.
5. Enter in a `Fiat` amount greater than your balance, you should see an error and the `Sell with Ramp` button should be disabled.
6. Enter in an amount under your balance and click `Sell with Ramp`
7. You should be directed to https://buy.ramp.network/ with `Sell` auto selected for you.
8. Your `Address`, `Fiat Value`, `Offramp Asset` and `Fiat Currency` should be prefilled for you.

https://user-images.githubusercontent.com/40611140/217718662-e611cecb-7727-4826-ae94-2b4c4392efca.mov

### Testing from Account
1. Go to the `Accounts` page and select an account.
2. If you have a balance for a supported token to sell, you will see a `Sell` button to the right of the asset.
3. Click on `Sell` and the `Sell` modal will display.
4. Check that the `HALF` and `MAX` buttons work.
5. Enter in a `Fiat` amount greater than your balance, you should see an error and the `Sell with Ramp` button should be disabled.
6. Enter in an amount under your balance and click `Sell with Ramp`
7. You should be directed to https://buy.ramp.network/ with `Sell` auto selected for you.
8. Your `Address`, `Fiat Value`, `Offramp Asset` and `Fiat Currency` should be prefilled for you.

https://user-images.githubusercontent.com/40611140/217719142-dc0cf70e-a8c2-4d87-9038-3abea43447e4.mov
